File: Napakalaki_Java/src/napakalaki/CultistPlayer.java
package napakalaki;

import java.util.Random;

public class CultistPlayer extends Player {
    
    static private int totalCultistPlayers = 0;
    private Cultist myCultistCard;
    
    public CultistPlayer(Player player, Cultist c) {
        super(player);
        this.myCultistCard = c;
        totalCultistPlayers++;
    }
    
    protected int getCombatLevel(){
        float nivel = super.getCombatLevel();
        nivel = (float) (nivel + (nivel*0.2));
        nivel += myCultistCard.getGainedLevels()*totalCultistPlayers;
        return (int) nivel;
    }
    
    protected int getOponentLevel(Monster m){
        return m.getCombatLevelAgainstCultistPlayer();
    }
    
    protected boolean shouldConvert(){
        return false;
    }

    private Treasure giveMeATreasure(){
        Random rand = new Random(); 
        return visibleTreasures.get(rand.nextInt(visibleTreasures.size()-1));
    }
    
    private boolean canYouGiveMeATreasure(){
        if(!visibleTreasures.isEmpty() || visibleTreasures != null)
            return true;
        else
            return false;
    }
    
    public static int getTotalCultistPlayers(){
        return totalCultistPlayers;
    }
}

File: Napakalaki_Ruby/lib/cultist_player.rb
=begin
  Marco Manuel Fernández Pranno, 2015 (Granada)
  Para la asignatura PDOO (Universidad de Granada)
  Licencia GPLv3 a partir del 15/02/16, hasta entonces todos los derechos reservados.
=end 

require_relative 'player'

class CultistPlayer < Player
  
  @@totalCultistPlayers = 0
  
  def initialize(p, c)
    super(nil,p)
    @myCultistCard = c
    @@totalCultistPlayers += 1
  end
  
  def getCombatLevel()
    nivel = @level
    nivel = (nivel + (nivel*0.2))
    nivel += @myCultistCard.gainedLevels*@@totalCultistPlayers
    return nivel.to_i
  end
  
  def getOponentLevel(m)
    return m.getCombatLevelAgainstCultistPlayer
  end
  
  def shouldConvert()
    return false
  end
  
  def giveMeATreasure()
    return @visibleTreasures[rand(@visibleTreasures.size-1)]
  end
  
  def canYouGiveMeATreasure()
    if( @visibleTreasures != nil || ! @visibleTreasures.empty?)
      return true
    end
    return false
  end
  
  def getTotalCultistPlayers
    return @@totalCultistPlayers
  end
end
